src/wallet/scriptpubkeyman.h

```diff
@@ -442,4 +442,20 @@ class LegacyScriptPubKeyMan : public ScriptPubKeyMan, public FillableSigningProv
     std::set<CKeyID> GetKeys() const override;
 };
 
+/** Wraps a LegacyScriptPubKeyMan so that it can be returned in a new unique_ptr */
+class LegacySigningProvider : public SigningProvider
+{
+private:
+    const LegacyScriptPubKeyMan& spk_man;
+public:
+    LegacySigningProvider(const LegacyScriptPubKeyMan& spk_man) : spk_man(spk_man) {}
```

Comment:
Call it `m_spk_man` and avoid `a(a)` → `m_spk_man(spk_man)` ?

Reply:
Done